### PR TITLE
feat: support cert-manager

### LINKED ISSUE #548: feat: Integrate cert-manager to manage secrets
Use [cert-manager](https://cert-manager.io/) to manager secrets, so we can dynamically update certificates, such as SSL certificates.

### PR DESCRIPTION
**Pull request by lingsamuel**
Please answer these questions before submitting a pull request

Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
Fixes #548

___
### New feature or improvement
- Support manage secrets with cert-manager

APISIX ingress controller now only supports secerts with `cert` and `key` fields, which is not Kubernetes style. Secrets created by cert-manager contain the standrd k8s-style fields `ca.crt`, `tls.crt` and `tls.key`.

This PR makes APISIX ingress controller to support both old field names and k8s standard field names, which allows APISIX ingress controller to work with cert-manager.

### CODE REVIEW

File: docs/en/latest/practices/manage-certificates-with-cert-manager.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+This tutorial will detail how to manage secrets of ApisixTls using cert-manager.
+
+## Prerequisites
+
+* Prepare an available Kubernetes cluster in your workstation, we recommend you to use [Minikube](https://github.com/kubernetes/minikube).
```
Comment (gxthrj) [inline docs/en/latest/practices/manage-certificates-with-cert-manager.md:28]: How about recommend to use kind.

File: pkg/ingress/secret.go [hunk 2]
```diff
@@ -143,54 +142,58 @@ func (c *secretController) sync(ctx context.Context, ev *types.Event) error {
 		}
 		tls, err := c.controller.apisixTlsLister.ApisixTlses(tlsNamespace).Get(tlsName)
 		if err != nil {
-			log.Warnw("secret related ApisixTls resource not found, skip",
+			log.Debugw("secret related ApisixTls resource not found, skip",
```
Comment (gxthrj) [inline pkg/ingress/secret.go:145]: I think the `warn` level is better.

File: pkg/ingress/secret.go [hunk 3]
```diff
@@ -143,54 +142,58 @@ func (c *secretController) sync(ctx context.Context, ev *types.Event) error {
...
+					c.controller.recordStatus(tls, _resourceSyncAborted, err, metav1.ConditionFalse)
+				}(tls)
 				return true
 			}
 			ssl.Client = &apisixv1.MutualTLSClientConfig{
 				CA: string(ca),
 			}
 		} else {
-			log.Warnw("stale secret cache, ApisixTls doesn't requires target secret",
+			log.Infow("stale secret cache, ApisixTls doesn't requires target secret",
```
Comment (gxthrj) [inline pkg/ingress/secret.go:188]: Ditto

File: pkg/ingress/secret.go [hunk 4]
```diff
@@ -143,54 +142,58 @@ func (c *secretController) sync(ctx context.Context, ev *types.Event) error {
...
 				zap.String("ApisixTls", tlsMetaKey),
 				zap.String("secret", key),
 			)
 			return true
 		}
 		// Use another goroutine to send requests, to avoid
 		// long time lock occupying.
-		go func(ssl *apisixv1.Ssl) {
+		go func(ssl *apisixv1.Ssl, tls *configv1.ApisixTls) {
```
Comment (gxthrj) [inline pkg/ingress/secret.go:196]: Suggest to add `context.WithTimeout`.

Comment (lingsamuel) [reply pkg/ingress/secret.go:196]: How long is reasonable?

Comment (gxthrj) [reply pkg/ingress/secret.go:196]: My bad, In fact, `ctx` is already used in the next line.

**Review by gxthrj — APPROVED** [review]

File: docs/en/latest/practices/manage-certificates-with-cert-manager.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+This means that our cert-manager is working properly.
+
+## Create Test Service
+
+We use [kennethreitz/httpbin](https://hub.docker.com/r/kennethreitz/httpbin/) as the service image.
+
+Deploy it by running:
+
+```bash
+kubectl run httpbin --image kennethreitz/httpbin --port 80
```
Comment (tao12345666333) [inline docs/en/latest/practices/manage-certificates-with-cert-manager.md:121]:
```suggestion
kubectl run httpbin --image kennethreitz/httpbin --expose --port 80
```

We can complete the creation of pod and svc directly with one command

File: docs/en/latest/practices/manage-certificates-with-cert-manager.md [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+kubectl expose pod httpbin --port 80
```
Comment (tao12345666333) [inline docs/en/latest/practices/manage-certificates-with-cert-manager.md:122]: And remove this line.

File: docs/en/latest/practices/manage-ingress-certificates-with-cert-manager.md [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+This means that our cert-manager is working properly.
+
+## Create Test Service
+
+We use [kennethreitz/httpbin](https://hub.docker.com/r/kennethreitz/httpbin/) as the service image.
+
+Deploy it by running:
+
+```bash
+kubectl run httpbin --image kennethreitz/httpbin --port 80
```
Comment (tao12345666333) [inline docs/en/latest/practices/manage-ingress-certificates-with-cert-manager.md:117]: As I suggested above.

**Review by tao12345666333 — APPROVED** [review]
LGTM!!!

**Pull request merged**